staging/src/k8s.io/client-go/tools/record/event.go

```diff
@@ -172,8 +172,8 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 func recordToSink(sink EventSink, event *v1.Event, eventCorrelator *EventCorrelator, sleepDuration time.Duration) {
 	// Make a copy before modification, because there could be multiple listeners.
 	// Events are safe to copy like this.
-	eventCopy := *event
-	event = &eventCopy
+	eventCopy := event.DeepCopy()
```

Comment:
1) bug fixes need to come with a test demonstrating the problem, to ensure there is no regression in the future.
2) please don't leave stale comments.

Reply:
@lavalamp 
I looked at sink and event correlator.
None of them provides accessor to the event they process.

If you have idea how to retrieve the new event (see recordEvent), please let me know.

Meanwhile, I will continue to investigate.

Reply:
Tried the following:
```
diff --git a/staging/src/k8s.io/client-go/tools/record/event_test.go b/staging/src/k8s.io/client-go/tools/record/event_test.go
index eadb1331c9..3bd57b73fa 100644
--- a/staging/src/k8s.io/client-go/tools/record/event_test.go
+++ b/staging/src/k8s.io/client-go/tools/record/event_test.go
@@ -436,9 +436,23 @@ func TestWriteEventError(t *testing.T) {
        }
 }

+type testEventCorrelator struct {
+       eventCorrelator *EventCorrelator
+       event *v1.Event
+}
+func (ec *testEventCorrelator) EventCorrelate(newEvent *v1.Event) (*EventCorrelateResult, error) {
+       ec.event = newEvent.DeepCopy()
+       return ec.eventCorrelator.EventCorrelate(newEvent)
+}
+func (ec *testEventCorrelator) UpdateState(event *v1.Event) {
+       ec.UpdateState(event)
+}
+
 func TestUpdateExpiredEvent(t *testing.T) {
        clock := clock.IntervalClock{Time: time.Now(), Duration: time.Second}
+       tec := testEventCorrelator{}
        eventCorrelator := NewEventCorrelator(&clock)
+       tec.eventCorrelator = eventCorrelator

        var createdEvent *v1.Event

@@ -459,7 +473,7 @@ func TestUpdateExpiredEvent(t *testing.T) {
        ev := &v1.Event{}
        ev.ResourceVersion = "updated-resource-version"
        ev.Count = 2
-       recordToSink(sink, ev, eventCorrelator, 0)
+       recordToSink(sink, ev, &tec, 0)

        if createdEvent == nil {
                t.Error("Event did not get created after patch failed")
```
which doesn't compile because &tec is not *EventCorrelator

Reply:
Turned out to be easier than I initially thought.

I have updated the test with reflect.DeepEqual (which is stricter than previous assertion).